File: discovery-service/src/main/java/com/sudo248/discoveryservice/controller/dto/ProductDto.java
package com.sudo248.discoveryservice.controller.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductDto {
    private String productId;
    private String name = "",description = "",sku = "";
    private List<ImageDto> images = new ArrayList<>();
    private List<String> categoryIds = new ArrayList<>();
    private List<SupplierProductDto> supplierProducts = new ArrayList<>();

    public ProductDto(String productId, String name, String description, String sku, List<ImageDto> images) {
        this.productId = productId;
        this.name = name;
        this.description = description;
        this.sku = sku;
        this.images = images;
    }
}

File: client/Soc/app/src/main/java/com/sudo248/soc/data/dto/discovery/ProductDto.kt
package com.sudo248.soc.data.dto.discovery

data class ProductDto(
    val productId: String,
    val name: String,
    val description: String,
    val sku: String,
    val images: List<ImageDto>? = null,
    val categoryIds: List<String>,
    val supplierProducts: List<SupplierProductDto>,
)